### ISSUE TITLE
How to reuse vscode extension sources

### ISSUE THREAD

**Issue by kameshsampath** [Azure/vscode-kubernetes-tools#258]
I searched and asked questions in the Stackoverflow, is there a way we could reuse the extension dependency sources, right now I see there is no way :( .  

sorry for being ignorant here for node we usually the get modules and start reusing the code and import the relavant sources. guess there is no way with vscode extension plugin sources. 

Any way tips/tricks to do it?

**Comment by itowlson** [comment]
VS Code doesn't support source-level import of other extensions a la Node.js / npm imports.  However, it does allow extensions to expose an API, and we are actively planning to do that for the Kubernetes extension.  This would mean your extension could declare a dependency on this extension and make use of the services exposed via the API.  We're gathering requirements / use cases on the extension wiki (https://github.com/Azure/vscode-kubernetes-tools/wiki/Extensibility-discussion) - please tell us more about what you're trying to do either here or there (or contact me privately on `<same_as_github_username>@microsoft.com`), and we'll try to serve you.  Let's talk!

If you find you really need/want to call into our source code, or if we can't give you a suitable API in a timely manner, I think you would need to copy the relevant code into your repo by hand.  Please check our licence but I think it's pretty permissive (an acknowledgement and link would be nice though).

cc @squillace

**Comment by bnookala** [comment]
@itowlson we could also package utilities into their own node module - and publish that.

**Comment by itowlson** [comment]
@bnookala Yes, and that would force us to finally sort out our shambolic jumble of tool invocation systems... **grin**
